corehq/form_processor/models/forms.py

```diff
@@ -0,0 +1,417 @@
+from django.db import models
+
+from jsonfield.fields import JSONField
+from lxml import etree
+from memoized import memoized
+
+from couchforms import const
+from couchforms.jsonobject_extensions import GeoPointProperty
+from dimagi.ext import jsonobject
+from dimagi.utils.couch import RedisLockableMixIn
+from dimagi.utils.couch.safe_index import safe_index
+from dimagi.utils.couch.undo import DELETED_SUFFIX
+
+from corehq.blobs.exceptions import NotFound
+from corehq.sql_db.models import PartitionedModel, RequireDBManager
+
+from ..exceptions import AttachmentNotFound, MissingFormXml, XFormNotFound
+from ..track_related import TrackRelatedChanges
+from .abstract import AbstractXFormInstance
+from .attachment import AttachmentMixin
+from .mixin import SaveStateMixin
+
+
+class XFormInstanceManager(RequireDBManager):
+
+    def get_form(self, form_id, domain=None):
+        """Get form in domain
+
+        This will get a form from any domain if the domain is not provided,
+        but that is deprecated and should be avoided.
```

Comment:
nit: I wouldn't say it's deprecated since we still rely on that functionality in certain places.

Reply:
54efb53f9bec711516b05a224b60e3567532b419